require.js

```diff
@@ -87,11 +83,20 @@
     if (typeof bundle === 'object') {
       for (var key in bundle) {
         if (has.call(bundle, key)) {
-          modules[key] = bundle[key];
+          require.register(modules[key], bundle[key]);
         }
       }
     } else {
       modules[bundle] = fn;
+      if (extExp.test(bundle)) {
+        var woExt = bundle.replace(extExp, '');
+        var existing = aliases[woExt];
+        if (!existing || existing !== woExt + '.js') aliases[woExt] = bundle;
+      }
+      if (indexExp.test(bundle)) {
+        var woIndex = bundle.replace(indexExp, '');
+        if (!has.call(aliases, woIndex)) aliases[woIndex] = bundle;
+      }
```

Comment:
* alias `foo/bar.js` to `foo/bar` and `foo/bar/index.js` to `foo/bar/index`.
* alias `foo/bar.html` to `foo/bar` unless `foo/bar.js` is aliased there.
* alias `foo/bar/index.js` to `foo/bar` unless `foo/bar.js` is aliased there.